src/openrct2/scripting/ScTile.hpp

```diff
@@ -491,6 +497,82 @@ namespace OpenRCT2::Scripting
             }
         }
 
+        bool isHidden_get() const
+        {
+            // TODO: Simply return the 'hidden' field once corrupt elements are superseded.
+            const TileElement* element = map_get_first_element_at(_coords);
+            bool previousElementWasUsefulCorrupt = false;
+            do
+            {
+                if (element == _element)
+                    return previousElementWasUsefulCorrupt;
+
+                if (element->GetType() == TILE_ELEMENT_TYPE_CORRUPT)
+                    previousElementWasUsefulCorrupt = !previousElementWasUsefulCorrupt;
+                else
+                    previousElementWasUsefulCorrupt = false;
+            } while (!(element++)->IsLastForTile());
+
+            Guard::Assert(false);
+            return false;
+        }
+        void isHidden_set(bool hide)
+        {
+            // TODO: Simply update the 'hidden' field once corrupt elements are superseded.
+            ThrowIfGameStateNotMutable();
+            const bool isHidden = isHidden_get();
+            if (hide == isHidden)
+                return;
+
+            if (hide)
+            {
+                // Get index of our current element (has to be done now before inserting the corrupt element)
+                const auto elementIndex = _element - map_get_first_element_at(_coords);
+
+                // Insert corrupt element at the end of the list for this tile
+                // Note: Z = 255 guarantee this
+                TileElement* insertedElement = tile_element_insert({ _coords, 0xFF }, 0);
```

Comment:
Could you create a constant for this use of 0xFF. It is also used for discarded elements.